x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,199 @@
+  autoIntervalEnabled,
+  maxBars,
+  interval,
+  maxHistogramBars,
+  onToggleEditor,
+  toggleAutoInterval,
+  onMaxBarsChange,
+  onIntervalChange,
+}: {
+  autoIntervalEnabled: boolean;
+  maxBars: 'auto' | number;
+  interval: number;
+  maxHistogramBars: number;
+  onToggleEditor: () => void;
+  toggleAutoInterval: (enabled: boolean) => void;
+  onMaxBarsChange: (newMaxBars: number) => void;
+  onIntervalChange: (newInterval: number) => void;
+}) => {
+  // TODO: manage the temporary empty string for interval
+  const sectionLabel = i18n.translate('xpack.lens.indexPattern.ranges.granularity', {
+    defaultMessage: 'Granularity',
+  });
+
+  return (
+    <>
+      <EuiFormRow
+        label={sectionLabel}
+        data-test-subj="indexPattern-ranges-section-label"
+        labelAppend={
+          <EuiSwitch
+            label={i18n.translate('xpack.lens.indexPattern.ranges.autoInterval', {
+              defaultMessage: 'Auto interval',
+            })}
+            checked={autoIntervalEnabled}
+            onChange={(e) => toggleAutoInterval(e.target.checked)}
+            data-test-subj="indexPattern-ranges-auto-interval"
+            compressed
+          />
+        }
+      >
+        <>
+          <EuiFlexGroup gutterSize="s" responsive={false} wrap>
+            <EuiFlexItem>
+              {autoIntervalEnabled ? (
+                <EuiRange
+                  min={1}
+                  max={maxHistogramBars}
+                  step={1}
+                  value={maxBars === 'auto' ? '' : maxBars}
+                  onChange={({ target }) =>
+                    onMaxBarsChange(Number(get(target, 'value', maxHistogramBars)))
+                  }
+                  placeholder={i18n.translate('xpack.lens.indexPattern.ranges.autoIntervals', {
+                    defaultMessage: 'Auto',
+                  })}
+                  showLabels
+                  showInput="inputWithPopover"
+                  prepend={
+                    <>
+                      <EuiText size="s">
+                        {i18n.translate('xpack.lens.indexPattern.ranges.maxIntervals', {
+                          defaultMessage: 'Max intervals',
+                        })}
+                      </EuiText>{' '}
+                      <EuiIconTip
+                        position="right"
+                        content={i18n.translate('xpack.lens.indexPattern.ranges.maxIntervalsHelp', {
+                          defaultMessage:
+                            "Intervals will be selected automatically based on the available data. The maximum number of bars can never be greater than the Advanced Setting's {histogramMaxBars}",
+                          values: { histogramMaxBars: UI_SETTINGS.HISTOGRAM_MAX_BARS },
+                        })}
+                        type="questionInCircle"
+                      />
+                    </>
+                  }
+                />
+              ) : (
+                <EuiFieldNumber
```

Comment:
Remember that all form controls in popovers should be compressed
```suggestion
                <EuiFieldNumber
                  compressed
```